### PR TITLE
Add profiler autostart indicator to EditorRunBar

### PR DESCRIPTION
**Pull request by Geometror**
Followup to #96759

Addresses concerns that making the autostart of profilers persistent between sessions could lead to unexpected performance issues if users forget it's enabled.

Adds a small indicator near the EditorRunBar which shows up when autostart is enabled for any profiler.
Hovering shows the tooltip below.
Clicking opens the debugger view of the bottom panel and selects the first profiler for which autostart is enabled.

{08AD3E20-A18B-4A22-A8BB-324E6FC4EBC7}

### CODE REVIEW

**Comment by arkology** [comment]
Just some thoughts.
Maybe making this button yellow color is too much. Maybe something more neutral? I mean, you enable autostart and will be always "warned" even if it is inside single session.
How about to show it only when launching project and running it? And with less distractive color.

**Comment by akien-mga** [comment]
Looks pretty good to me overall!

I agree it might seem too distracting in the long run though, especially if some users want to always profile and are fine with the performance impact. But I'm also fine test driving it as is and seeing if users ask for changes.

**Comment by FoxyFox909** [comment]
> Just some thoughts. Maybe making this button yellow color is too much (may look like "something terrible happened, you forgot to turn off the kettle"😄). Maybe something more neutral? I mean, you enable autostart and will be always "warned" even if it is inside single session. How about to show it only when launching project and running it? And with less "distractive" color. For not to warn but notify.

Maybe a softer yellow sure, but I agree with yellow in general since yellow = warning, and red = error in Godot already

**Comment by KoBeWi** [comment]
I see some hard-coded values for the tab indices and profiler names. Ideally the profilers should register themselves to avoid hard-coding them. This can be refactored later though.

File: editor/gui/editor_run_bar.cpp [hunk 1]
```diff
@@ -353,6 +370,28 @@ bool EditorRunBar::is_movie_maker_enabled() const {
...
+void EditorRunBar::update_profiler_autostart_indicator() {
+	bool profiler_active = EditorSettings::get_singleton()->get_project_metadata("debug_options", "autostart_profiler", false);
+	bool visual_profiler_active = EditorSettings::get_singleton()->get_project_metadata("debug_options", "autostart_visual_profiler", false);
+	bool network_profiler_active = EditorSettings::get_singleton()->get_project_metadata("debug_options", "autostart_network_profiler", false);
+	bool any_profiler_active = profiler_active | visual_profiler_active | network_profiler_active;
+	profiler_autostart_indicator->set_visible(any_profiler_active);
+	if (any_profiler_active) {
+		String tooltip = TTR("Autostart is enabled for the following profilers, which can have a performance impact:");
+		if (profiler_active) {
+			tooltip += "\n- " + TTR("Profiler");
```
Comment (KoBeWi) [inline editor/gui/editor_run_bar.cpp:382]: Not important, but it's better to use PackedStringArray or StringBuilder instead.

**Review by KoBeWi — APPROVED** [review]
The implementation is ok, though as I mentioned, the hard-coding should be eventually removed.

**Comment by Geometror** [comment]
Regarding the button's color and style: Initially, I went with a yellow icon and a flat button, but it looked off and wasn’t noticeable, so I inversed it. I realize this might feel a bit bold to some, but it provides a clear visual cue that your brain associates with "profiling mode." This can also work in reverse - you quickly notice when the indicator is absent, helping you spot when autostart isn't active but wanted. As for the color, if you can give me some suggestions I can try those, but IMO the default-warning-yellow works quite well here.

@KoBeWi I forgot to add a note on the hardcoded values :) Yep, that felt horrible, but I fear it's currently the only way to achieve this. (however, I've added this to my profiler refactoring list)

**Comment by akien-mga** [comment]
Needs rebase.

**Comment by Geometror** [comment]
Rebased.

**Review by akien-mga — APPROVED** [review]

**Comment by Repiteo** [comment]
Thanks!

**Comment by WhalesState** [comment]
Just saw many gradients in the svg code and decided to have a look to remove them since it's using a single color and have found those designs xD

image

**Pull request merged**